examples/webgl_custom_clipping.html

```diff
@@ -0,0 +1,399 @@
+					mesh.matrixAutoUpdate = false;
+
+					volumeVisualization.add( mesh );
+
+				}
+
+				scene.add( volumeVisualization );
+
+
+				var ground = new THREE.Mesh( planeGeometry,
+						new THREE.MeshPhongMaterial( {
+							color: 0xa0adaf, shininess: 150 } ) );
+				ground.rotation.x = - Math.PI / 2;
+				ground.scale.multiplyScalar( 3 );
+				ground.receiveShadow = true;
+				scene.add( ground );
+
+				// Renderer
+
+				var container = document.body;
+
+				renderer = new THREE.WebGLRenderer();
+				renderer.shadowMap.enabled = true;
+				renderer.setPixelRatio( window.devicePixelRatio );
+				renderer.setSize( window.innerWidth, window.innerHeight );
+				window.addEventListener( 'resize', onWindowResize, false );
+				container.appendChild( renderer.domElement );
+				// Clipping setup:
+				globalClippingPlanes = createPlanes( GlobalClippingPlanes.length );
+				renderer.clippingPlanes = Empty;
+				renderer.localClippingEnabled = true;
+
+				// Stats
+
+				stats = new Stats();
+				container.appendChild( stats.domElement );
+
+				// Controls
+
+				var controls = new THREE.OrbitControls( camera, renderer.domElement );
+				controls.target.set( 0, 1, 0 );
+				controls.update();
+
+				// GUI
+
+				var gui = new dat.GUI(),
+					folder = gui.addFolder( "Local Clipping" ),
+					props = {
+						get 'Enabled'() { return renderer.localClippingEnabled; },
+						set 'Enabled'( v ) {
+								renderer.localClippingEnabled = v;
+								if ( ! v ) volumeVisualization.visible = false; },
+
+						get 'Shadows'() { return clipMaterial.clipShadows; },
+						set 'Shadows'( v ) { clipMaterial.clipShadows = v; },
+
+						get 'Visualize'() { return volumeVisualization.visible; },
+						set 'Visualize'( v ) {
+								if ( renderer.localClippingEnabled )
+									volumeVisualization.visible = v; }
+					};
+
+				folder.add( props, 'Enabled' );
+				folder.add( props, 'Shadows' );
+				folder.add( props, 'Visualize' ).listen();
+
+				gui.addFolder( "Global Clipping" ).
+						add( {
+							get 'Enabled'() { return renderer.clippingPlanes !== Empty; },
+							set 'Enabled'( v  ) { renderer.clippingPlanes = v ?
+									globalClippingPlanes : Empty; }
+						}, "Enabled" );
+
+
+				// Start
+
+				startTime = Date.now();
+
+			}
+
+			function onWindowResize() {
+
+				camera.aspect = window.innerWidth / window.innerHeight;
+				camera.updateProjectionMatrix();
+
+				renderer.setSize( window.innerWidth, window.innerHeight );
+
+			}
+
+			var transform = new THREE.Matrix4(),
+				tmpMatrix = new THREE.Matrix4();
+
+			function animate() {
+
+				var currentTime = Date.now(),
+					time = ( currentTime - startTime ) / 1000;
+
+				requestAnimationFrame( animate );
+
+				transform.makeTranslation( 0, 1, 0 ).
+						multiply( tmpMatrix.makeRotationX( time * 0.5 ) ).
+						multiply( tmpMatrix.makeRotationY( time * 0.2 ) );
+
+				object.matrix.copy( transform );
+				object.updateMatrixWorld( true );
```

Comment:
>Keep in mind that the planes are specified in world space

Ah, yes, of course. Even for material-specific clip planes. For some reason I was thinking they were in local space.

> happy to find a debugging tool in the example code :).

That is a good, actually. Maybe `THREE.PlaneHelper` is in order?

>I also think you might be underestimating the average Three.js user. 

After answering ~ 1,000 three.js questions on stackoverflow, I do know that many users find the three.js examples too complicated to follow. Granted, that may be a biased sample, but I do favor simpler examples, in lieu of showcases. JMHO.